### PR TITLE
i18n

### PR DESCRIPTION
**Pull request by fisx**
Related: #788.  This is just part of the solution, though.  Missing:

- establish transifex workflow
- factor this out into its own package (for once, that will make bootstrapping easier)
- support translations in js code (use yesod's `juslius`?)
- support translations for error messages etc. in `Action`.
- can we find a way to get rid of passing the language explicitly, at least in some cases?  (does it help to replace the functions `t_*` by constructors `T_*` that instantiate different type classes for different use cases?)

### CODE REVIEW

**Comment by fisx** [comment]
can we work with this as a transifex file format?:

```json
{
    "src/Frontend/Page/Topic.hs": {
        "t_shift_phase_forward:198": "Nächste Phase",
        "t_shift_phase_backward:206": "Vorherige Phase",
        "t_2:255": "window.location = this.value",
        "t_1:255": "heroic-tabs-dropdown"
    },
    "src/Frontend/Page/Idea.hs": {
        "t_bearbeiten:259": "*** t_bearbeiten",
        "t_wef:260:291": "*** t_wef"
    }
}
```

File: exec/I18N.hs [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+                <- runGit ["diff", "--word-diff=porcelain", "--no-color", cs gitfrom <> ".." <> cs gitto]
+            raw <- ST.hGetContents dout
+            _ <- ST.length raw `seq` waitForProcess dh
+            pure raw
+
+    pure $ fromTransKeys tkeys
+
+runGit :: [String] -> IO (Handle, Handle, Handle, ProcessHandle)
+runGit args = do
+    -- putStrLn . unwords $ "git" : (show <$> args)
```
Comment (andorp) [inline exec/I18N.hs]: Leftover `putStrLn` please remove if this line is not intended.

File: exec/I18N.hs [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+      gotf (TransTableF m) = Aeson.object . fmap (uncurry (Aeson..=) . (_1 %~ cs)) . Map.toList . Map.map gotk $ m
+
+      gotk :: TransTableK -> Aeson.Value
+      gotk (TransTableK m) = Aeson.object $ uncurry gott <$> Map.toList m
+
+      gott :: TKey -> TransTableT -> Aeson.Pair
+      gott k (TransTableT (ls, m)) = ak Aeson..= av
+        where
+          ak = k <> ":" <> ST.intercalate ":" (cs . show <$> Set.toAscList ls)
+          av = fromMaybe (error "Transifex: incomplete translation table") $ Map.lookup lang m
```
Comment (andorp) [inline exec/I18N.hs]: The `lang` parameter should be included in the message to give the user some context.

File: exec/I18N.hs [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+        putStrLn $ "updating " <> show translationsFile <> "..."
+        LBS.writeFile translationsFile (Aeson.encodePretty table')
+
+        putStrLn "generating `*_T` modules..."
+        compileTransTableF table'
+
+        putStrLn "done!"
+
+
+-- TODO: make this a separate package thentos-i18n
```
Comment (andorp) [inline exec/I18N.hs]: Is this `TODO` still relevant?

**Comment by andorp** [comment]
This one is a big addition. I am a little bit concerned about the dense naming, but there is no big probability that the code is going to change, so I fine with it.

**Comment by andorp** [comment]
LGTM

**Comment by andorp** [comment]
Wait for travis.

**Comment by andorp** [comment]
LGTM

**Pull request merged**